src/parser/statement.js

```diff
@@ -668,7 +668,6 @@ pp.parseClassBody = function (node) {
 
     method.static = isMaybeStatic && !this.match(tt.parenL);
     if (method.static) {
-      if (isGenerator) this.unexpected();
```

Comment:
Yeah, `method.static` hinges on `this.match(tt.name)` and `isGenerator` (subsequently) on `this.eat(tt.star)`. So they are mutually exclusive.